src/transformers/generation/utils.py

```diff
@@ -2379,7 +2379,7 @@ def _dola_decoding(
             for candidate_premature_layer in candidate_premature_layers:
                 candidate_premature_logits[candidate_premature_layer] = lm_head(
                     outputs.hidden_states[candidate_premature_layer][:, -1, :]
-                )
+                ).to(final_logits.device)
```

Comment:
We need to run ops using both `final_logits` (the logits from the model, device=same device as the model input) and `candidate_premature_logits` (tensor computed using `lm_head`, device = same device as `lm_head`).

Given that the `logits` is a well-established concept in `transformers`, the tensor that should be moved (IMO) is the other one :)